python/GafferSceneUI/LightEditor.py

```diff
@@ -444,6 +451,32 @@ def __activeInspectionTweaks( self, pathListing ) :
 
 		return tweaks
 
+	def __removableAttributeTweaks( self, pathListing ) :
+
+		tweaks = []
+
+		with Gaffer.Context( self.getContext() ) as context :
+			for columnSelection, column in zip( pathListing.getSelection(), pathListing.getColumns() ) :
+				if not isinstance( column, _GafferSceneUI._LightEditorInspectorColumn ) :
+					continue
+				for path in columnSelection.paths() :
+					context["scene:path"] = GafferScene.ScenePlug.stringToPath( path )
+					inspection = column.inspector().inspect()
+					if (
+						inspection is not None and
+						inspection.editable() and
+						inspection.sourceType() == GafferSceneUI.Private.Inspector.Result.SourceType.EditScope
+					) :
+						edit = inspection.acquireEdit()
+						if (
+							isinstance( column.inspector(), GafferSceneUI.Private.AttributeInspector ) and
+							isinstance( edit, Gaffer.TweakPlug ) and
+							edit["mode"].getValue() != Gaffer.TweakPlug.Mode.Remove
+						) :
+							tweaks.append( edit )
+
+		return tweaks
```

Comment:
This seems to mean that we can only delete attributes in an EditScope if they have already been edited in that scope. I'd like to be able to delete _any_ attribute in an EditScope, by authoring a _new_ tweak as necessary. Is there any reason we can't do that?

Reply:
Yes, that's definitely a good change to make. Removing the test of `inspection.sourceType() == GafferSceneUI.Private.Inspector.Result.SourceType.EditScope` allows the edit scope to delete any attribute.

It also allows you to delete (set the tweak mode to `Remove` to be more precise) an attribute that doesn't exist. I'm not sure if that's desirable or not. I think perhaps it is, with the use-case being "I'm working on a shot setup and want to make sure this attribute is removed, even if someone upstream of my shot adds it in the future." Do you think that's worth keeping?

Reply:
I reworked `__removableAttributeTweaks()` in https://github.com/GafferHQ/gaffer/pull/4948/commits/31a45171fb615016ea10ba86e2e8e2485154a14f and I think it's looking much better now. And by returning an `inspector` list instead of plugs, we can avoid creating edits until they are really needed, not just from checking to see if the menu item is enabled.

Reply:
Yep - that seems like a nice way of approaching it. I think we need to do the same "is the whole selection removable?" check in the <kbd>Delete</kbd> shortcut as well though. Perhaps it would be even better if `__removableAttributeInspections` simply returned an empty list if it found anything that wasn't removable? Then the check is centralised and the mistake can't be repeated elsewhere...